syn/python/get_kge.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python3
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Parse a yosys area report and give a kGE equivalient
+
+import re
+import argparse
+
+weighted_dict = {}
+
+
+def read_lib(lib_file_path):
+    with open(lib_file_path, 'r') as f:
+        lib_file = f.readlines()
+
+    cell_dict = {}
+    cell_name = ''
+    for line in lib_file:
+        if re.search(r'^  cell \(', line):
+            if cell_name != '':
+                print("Size missing")
+                quit()
+            cell_name = line.split()[1].strip('()')
+        elif re.search(r'^\s+area', line):
+            if cell_name == '':
+                print("Size found without cell")
+                quit()
+            cell_area = line.split()[2].strip(';')
+            cell_dict[cell_name] = float(cell_area)
```

Comment:
Maybe add some error handling here? I don't think you know that `line.split()` will generate at least 3 words, and you don't know that `cell_area` is a valid number.

Reply:
I'll have a go at this - let me know if I've done it wrong